### PR TITLE
Add etcd CLI to arkade

### PR DESCRIPTION
**Pull request by klutchell**
## Description

Add etcd CLI to arkade

## Motivation and Context

Existing install process is clunky.

Before (from [here](https://docs.k3s.io/advanced#using-etcdctl) and [here](https://etcd.io/docs/v3.5/install/))

```bash
ETCD_VERSION="v3.5.5"
ETCD_URL="https://github.com/etcd-io/etcd/releases/download/${ETCD_VERSION}/etcd-${ETCD_VERSION}-linux-amd64.tar.gz"
curl -sL ${ETCD_URL} | sudo tar -zxv --strip-components=1 -C /usr/local/bin
```

After

```
arkade get etcd
```

Fixes: https://github.com/alexellis/arkade/issues/945

- [ ] I have raised an issue to propose this change, which has been given a label of `design/approved` by a maintainer ([required](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?

Unit tests, plus test tool output:

```bash
+ ./arkade get etcd --arch arm64 --os darwin --quiet
+ file /Users/kyle/.arkade/bin/etcd
/Users/kyle/.arkade/bin/etcd: Mach-O 64-bit arm64 executable, flags:<|DYLDLINK|PIE>
+ rm /Users/kyle/.arkade/bin/etcd
+ echo

+ ./arkade get etcd --arch x86_64 --os darwin --quiet
+ file /Users/kyle/.arkade/bin/etcd
/Users/kyle/.arkade/bin/etcd: Mach-O 64-bit x86_64 executable
+ rm /Users/kyle/.arkade/bin/etcd
+ echo

+ ./arkade get etcd --arch x86_64 --os linux --quiet
+ file /Users/kyle/.arkade/bin/etcd
/Users/kyle/.arkade/bin/etcd: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, Go BuildID=GyARklZhf9hi1aDRnamg/iBdV6hB-xGaEhWj0UVLe/sHhb4i0nu2-DUWKGC0Yw/4Y_A_aeQheHQIcJhyyC2, stripped
+ rm /Users/kyle/.arkade/bin/etcd
+ echo

+ ./arkade get etcd --arch aarch64 --os linux --quiet
+ file /Users/kyle/.arkade/bin/etcd
/Users/kyle/.arkade/bin/etcd: ELF 64-bit LSB executable, ARM aarch64, version 1 (SYSV), statically linked, Go BuildID=0_hf4eGUgeIZHwoJkukE/Cx-_IKMnyYRk24L0CmSi/o1qIZwTPv6-DoUGsXTHQ/i0Kjvgrkc76OspFXHhTs, stripped
+ rm /Users/kyle/.arkade/bin/etcd
+ echo

+ ./arkade get etcd --arch x86_64 --os mingw --quiet
+ file /Users/kyle/.arkade/bin/etcd.exe
/Users/kyle/.arkade/bin/etcd.exe: PE32+ executable (console) x86-64 (stripped to external PDB), for MS Windows, 6 sections
+ rm /Users/kyle/.arkade/bin/etcd.exe
+ echo
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Documentation

- [x] I have updated the list of tools in README.md if (required) with `./arkade get --format markdown`
- [ ] I have updated the list of apps in README.md if (required) with `./arkade install --help`

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`

- [x] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

File: pkg/get/get_test.go [hunk 1]
```diff
@@ -1133,6 +1133,58 @@ func Test_DownloadK9s(t *testing.T) {
...
+			url:     `https://github.com/etcd-io/etcd/releases/download/v3.4.27/etcd-v3.4.27-linux-amd64.tar.gz`,
+		},
+		{
+			os:      "darwin",
+			arch:    arch64bit,
+			version: toolVersion,
+			url:     `https://github.com/etcd-io/etcd/releases/download/v3.4.27/etcd-v3.4.27-darwin-amd64.zip`,
+		},
+		{
+			os:      "windows",
```
Comment (Jasstkn) [inline pkg/get/get_test.go:1158]: looks like duplication of windows test.

File: pkg/get/get_test.go [hunk 2]
```diff
@@ -1133,6 +1133,58 @@ func Test_DownloadK9s(t *testing.T) {
...
+
+	tests := []test{
+		{
+			os:      "linux",
+			arch:    arch64bit,
+			version: toolVersion,
+			url:     `https://github.com/etcd-io/etcd/releases/download/v3.4.27/etcd-v3.4.27-linux-amd64.tar.gz`,
+		},
+		{
+			os:      "darwin",
```
Comment (Jasstkn) [inline pkg/get/get_test.go:1152]: could you add a test for darwin arm?

**Review by Jasstkn — CHANGES_REQUESTED** [review]

File: pkg/get/get_test.go (hunk 2 above)
Comment (klutchell) [reply pkg/get/get_test.go:1152]: there are no binaries available for darwin arm unfortunately (e.g. https://github.com/etcd-io/etcd/releases/tag/v3.4.27)

Comment (klutchell) [reply pkg/get/get_test.go:1152]: is there a proper way to indicate in the template that a combination is not supported?

File: pkg/get/get_test.go (hunk 1 above)
Comment (klutchell) [reply pkg/get/get_test.go]: fixed!

File: pkg/get/get_test.go (hunk 2 above)
Comment (alexellis) [reply pkg/get/get_test.go:1152]: If there's no binary for an OS/arch mix, then we usually won't add a test.

Comment (alexellis) [reply pkg/get/get_test.go:1152]: cc @klutchell

Comment (klutchell) [reply pkg/get/get_test.go:1179]: My mistake, recent releases like v3.5.9 do include darwin arm64 binaries so I added the test cases and re-ran all tests.

**Review by Jasstkn — APPROVED** [review]
LGTM.

**Comment by alexellis** [comment]
Thanks for the review @Jasstkn I'll get this merged. Thanks also @klutchell

**Comment by alexellis** [comment]
https://github.com/alexellis/arkade/releases/tag/0.9.26

**Pull request merged**